packages/utils/upgrade/resources/codemods/5.0.0/useRBAC-hook-import-change.code.ts

```diff
@@ -0,0 +1,84 @@
+import type { Transform, ImportDeclaration, JSCodeshift, Collection } from 'jscodeshift';
+
+const changeImportSpecifier = (
+  root: Collection,
+  j: JSCodeshift,
+  options: { methodName: string; oldDependency: string; newDependency: string }
+): void => {
+  const { methodName, oldDependency, newDependency } = options;
+
+  // Flag to check if the method was imported from the old dependency
+  let methodImportedFromOldDependency = false;
+
+  // Remove the method from the old dependency and check if it was imported
+  root
+    .find(j.ImportDeclaration)
+    .filter((path) => path.node.source.value === oldDependency)
+    .forEach((path) => {
+      const importDeclaration: ImportDeclaration = path.node;
+
+      // Check if the method is imported from the old dependency
+      const methodExistsInOldDependency = importDeclaration.specifiers?.some(
+        (specifier) =>
+          specifier.type === 'ImportSpecifier' && specifier.imported.name === methodName
+      );
+
+      if (methodExistsInOldDependency) {
+        methodImportedFromOldDependency = true;
+
+        const updatedSpecifiers = importDeclaration.specifiers?.filter(
+          (specifier) =>
+            specifier.type === 'ImportSpecifier' && specifier.imported.name !== methodName
+        );
+
+        if (updatedSpecifiers && updatedSpecifiers.length > 0) {
+          // Replace the import with the updated specifiers if there are other imports left
+          j(path).replaceWith(j.importDeclaration(updatedSpecifiers, j.literal(oldDependency)));
+        } else {
+          // Remove the entire import statement if the specified method was the only import
+          j(path).remove();
+        }
+      }
+    });
+
+  // Add new import dependency if the method was imported from the old dependency
+  if (methodImportedFromOldDependency) {
+    const dependencies = root
+      .find(j.ImportDeclaration)
+      .filter((path) => path.node.source.value === newDependency);
+
+    dependencies.forEach((path) => {
+      const importDeclaration: ImportDeclaration = path.node;
+
+      const newSpecifier = j.importSpecifier(j.identifier(methodName));
+      importDeclaration.specifiers?.push(newSpecifier);
```

Comment:
```suggestion
      const importDeclaration: ImportDeclaration = path.node;

      const newSpecifier = j.importSpecifier(j.identifier(methodName));
      const specifiersArray = importDeclaration.specifiers || [];
      j(path).replaceWith(
        j.importDeclaration([...specifiersArray, newSpecifier], j.literal(newDependency))
      );
```
We can avoid mutating the AST by using `replaceWith`